File: taskManager/labfinal_210042111/ChangeIssueStatusCommand.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace labfinal_210042111
{
    public class ChangeIssueStatusCommand : ICommand
    {
        private IssueRepository repository;
        private Issue issue;
        private Status newStatus;
        private Status oldStatus;
        public ChangeIssueStatusCommand(IssueRepository repository, Issue issue, Status newStatus)
        {
            this.repository = repository;
            this.issue = issue;
            this.newStatus = newStatus;

        }
        public void Execute()
        {
            oldStatus = issue.getStatus();
            issue.changeStatus(newStatus);
        }

        public void undo()
        {
            issue.changeStatus(oldStatus);
        }
    }
}

File: TaskManager/LabFinal/src/commands/ChangeIssueStatusCommand.java
package commands;

import enums.Status;
import interfaces.ICommand;
import models.Issue;
import models.IssueRepository;

public class ChangeIssueStatusCommand implements ICommand {
    private final IssueRepository issueRepo;
    private final Issue issue;
    private final Status newStatus;
    private Status oldStatus;

    public ChangeIssueStatusCommand(IssueRepository repository, String issueId, Status newStatus) {

        this.issueRepo = repository;
        this.issue = issueRepo.getIssue(issueId);
        this.newStatus = newStatus;
    }

    @Override
    public void execute() {
        oldStatus = issue.getStatus();
        issue.setStatus(newStatus);
    }

    @Override
    public void undo() {
        if (oldStatus != null) {
            issue.setStatus(oldStatus);
        }
    }

}